--- systemtap-3.0/configure.ac.orig	2016-04-05 20:34:37.752195113 +0200
+++ systemtap-3.0/configure.ac	2016-04-05 20:44:11.672171031 +0200
@@ -93,8 +93,8 @@
 AS_IF([test "x$enable_ssp" != xno],[
    save_CFLAGS="$CFLAGS"
    save_CXXFLAGS="$CXXFLAGS"
-   CXXFLAGS="$CXXFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
-   CFLAGS="$CFLAGS -Werror -fstack-protector-all -D_FORTIFY_SOURCE=2"
+   CXXFLAGS="$CXXFLAGS -fstack-protector-all -D_FORTIFY_SOURCE=2"
+   CFLAGS="$CFLAGS -fstack-protector-all -D_FORTIFY_SOURCE=2"
    AC_COMPILE_IFELSE([AC_LANG_SOURCE([int something ();])], [
        AC_MSG_NOTICE([Compiling with gcc -fstack-protector-all et al.])
 	CFLAGS="$save_CFLAGS -fstack-protector-all -D_FORTIFY_SOURCE=2"
--- systemtap-3.0/Makefile.am.orig	2016-04-05 20:44:51.275502703 +0200
+++ systemtap-3.0/Makefile.am	2016-04-05 20:51:00.125487222 +0200
@@ -14,8 +14,8 @@
 	      -DLOCALEDIR='"$(localedir)"' \
 	      -I$(srcdir)/includes -I$(builddir)/includes/sys
 
-AM_CFLAGS = -D_GNU_SOURCE -fexceptions -Wall -Wextra -Werror -Wunused -Wformat=2 -W
-AM_CXXFLAGS = -Wall -Wextra -Werror
+AM_CFLAGS = -D_GNU_SOURCE -fexceptions -Wall -Wextra -Wunused -Wformat=2 -W
+AM_CXXFLAGS = -Wall -Wextra
 
 bin_SCRIPTS = stap-report
 pkglibexec_SCRIPTS = stap-env
