--- systemtap-1.8/Makefile.am~	2012-06-17 18:27:54.000000000 +0200
+++ systemtap-1.8/Makefile.am	2012-07-01 14:12:11.969612715 +0200
@@ -183,7 +183,7 @@
 stap_serverd_SOURCES = stap-serverd.cxx cscommon.cxx util.cxx privilege.cxx nsscommon.cxx cmdline.cxx
 stap_serverd_CXXFLAGS = $(AM_CXXFLAGS) @PIECXXFLAGS@ $(nss_CFLAGS)
 stap_serverd_CFLAGS = $(AM_CFLAGS) @PIECFLAGS@ $(nss_CFLAGS)
-stap_serverd_LDFLAGS = $(AM_LDFLAGS) @PIELDFLAGS@
+stap_serverd_LDFLAGS = $(AM_LDFLAGS) @PIELDFLAGS@ -pthread
 stap_serverd_LDADD = $(nss_LIBS)
 if HAVE_AVAHI
 stap_serverd_CFLAGS += $(avahi_CFLAGS)
