--- systemtap-5.1/configure.ac.orig	2024-04-26 23:34:34.000000000 +0200
+++ systemtap-5.1/configure.ac	2024-06-30 09:13:25.002462389 +0200
@@ -656,7 +656,7 @@ no) ;;
      old_CPPFLAGS="$CPPFLAGS"
      CPPFLAGS="$CPPFLAGS $DYNINST_CXXFLAGS"
      AC_LANG_PUSH(C++)
-     for location in /usr/lib64/dyninst /usr/lib/dyninst ${libdir}/dyninst; do
+     for location in ${libdir} /usr/lib64/dyninst /usr/lib/dyninst ${libdir}/dyninst; do
        LDFLAGS="-L${location}"
          LIBS="-lparseAPI ${BOOST_SYSTEM_LIB}"
 	   AC_LINK_IFELSE([AC_LANG_PROGRAM([[#include <dyninst/CodeSource.h>]],
@@ -678,7 +678,7 @@ no) ;;
        save_LIBS="$LIBS"
        AC_CHECK_LIB(tbb, TBB_runtime_interface_version, [tbb_LIBS=-ltbb],[tbb_LIBS=])       
        LIBS="$save_LIBS"
-       DYNINST_LIBS="-lparseAPI -lsymtabAPI -linstructionAPI $tbb_LIBS -lcommon"
+       DYNINST_LIBS="-lparseAPI -lsymtabAPI -linstructionAPI $tbb_LIBS -ldyncommon"
      fi
      ;;
 *)  # Use paths in the user-specified prefix
