--- systemtap-1.4/configure.ac~	2011-01-17 12:31:41.000000000 +0100
+++ systemtap-1.4/configure.ac	2011-05-17 11:16:21.590000069 +0200
@@ -548,6 +548,9 @@
   AC_CHECK_HEADERS([elfutils/version.h])
 fi
 
+CPPFLAGS="${CPPFLAGS} -I/usr/include/rpm"
+LIBS="${LIBS} -lrpmdb-4.5"
+
 dnl This is here mainly to make sure that configure --prefix=... changes
 dnl the config.h files so files depending on it are recompiled
 dnl prefix is passed through indirectly in the Makefile.am AM_CPPFLAGS.
