--- systemtap-5.1/httpd/server.cxx.orig	2024-04-26 23:34:34.000000000 +0200
+++ systemtap-5.1/httpd/server.cxx	2024-06-30 20:45:04.377640976 +0200
@@ -214,7 +214,7 @@ connection_info::postdataiterator(enum M
 				  size_t size)
 {
     if (filename && key) {
-	server_error(_F("key='%s', filename='%s', size=%ld", key, filename,
+	server_error(_F("key='%s', filename='%s', size=%zd", key, filename,
 			size));
 
 	// If we've got a filename, we need a temporary directory to
--- systemtap-5.4/elaborate.cxx.orig	2025-10-30 15:47:35.000000000 +0100
+++ systemtap-5.4/elaborate.cxx	2026-02-22 15:47:22.632541432 +0100
@@ -5116,7 +5116,7 @@ const_folder::visit_defined_op (defined_
       // Don't be greedy... we'll only collapse one at a time so type
       // resolution can have another go at it.
       relaxed_p = false;
-      session.print_warning (_F("Collapsing unresolved @define to %ld [stapprobes]", value), e->tok);
+      session.print_warning (_F("Collapsing unresolved @define to %" PRId64 " [stapprobes]", value), e->tok);
       literal_number* n = new literal_number (value);
       n->tok = e->tok;
       n->visit (this);
